### PR TITLE
[2198] Do not blindly set #threads to requested #threads (4-2-stable)

### PR DESCRIPTION
**Pull request by JustinKyleJames**
Test run of the main suite of tests passed.

### CODE REVIEW

File: packaging/resource_suite_s3_nocache.py [hunk 1]
```diff
@@ -2356,6 +2357,42 @@ def test_upload_very_large_file_after_redirect_issues_2104_2114(self):
...
+            for file_size in file_sizes:
+                # create the file
+                lib.make_arbitrary_file(filename, file_size)
+
+                # put the file
+                self.user1.assert_icommand('iput %s' % filename)
+                self.user1.assert_icommand('iget -f %s %s' % (filename, get_filename))
+                assert_command(['diff', '-q', filename, get_filename])
+
+                for n_value in [2,3,4,8]:
```
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2382]: do we want to add a 0 and 1 to this array?

Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: done

**Comment by trel** [comment]
if everything passes, please squash, no #.   and we'll wait for another set of eyes.

**Comment by JustinKyleJames** [comment]
> if everything passes, please squash, no #. and we'll wait for another set of eyes.

I added an additional test case for a recursive put with -N flag and checksum as that was initially reported as the issue.  It's a little overkill.

I squashed the one additional commit but kept this latest commit so you can take a look at it.

File: packaging/resource_suite_s3_nocache.py [hunk 2]
```diff
@@ -2393,6 +2394,31 @@ def test_iput_with_N_flag_issue_2198(self):
...
+        filename1 = '%s1' % dirname
+        filename2 = '%s2' % dirname
+        filename3 = '%s3' % dirname
+        try:
+            # create the local directory
+            os.mkdir(dirname)
+            # create the files
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename1), 100)
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename2), 100)
+            lib.make_arbitrary_file('%s/%s'  % (dirname, filename3), 100)
```
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2409]: double spaces before %

File: packaging/resource_suite_s3_nocache.py [hunk 3]
```diff
@@ -2393,6 +2394,31 @@ def test_iput_with_N_flag_issue_2198(self):
             # clean up the file that was put in case it wasn't cleaned up
             self.user1.assert_icommand('irm -f %s' % filename)
 
+    def recursive_put_small_files_with_checksum_and_N_flag_issue_2198(self):
+
+        dirname = 'recursive_put_small_files_with_checksum_and_N_flag_issue_2198'
```
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2399]: do we usually use "__issue"... in these names?

Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: Usually.  I made the change.

File: packaging/resource_suite_s3_nocache.py (hunk 2 above)
Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: Removed the extra space.

**Comment by alanking** [comment]
Are we planning to do a 4.2.12.2 release?

**Comment by korydraughn** [comment]
No due to 4.2.12 being EOL.

**Comment by JustinKyleJames** [comment]
> Are we planning to do a 4.2.12.2 release?

I think this is important enough to have a release in 4.2.12.

**Comment by trel** [comment]
Agreed.

**Comment by alanking** [comment]
Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

**Comment by JustinKyleJames** [comment]
> Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

I will make those changes on this one.  Unfortunately it is a manual process due to the python->python3 update.

I am going to hold off on making the changes for #2204 until that other PR gets approved and merged.

**Comment by JustinKyleJames** [comment]
I made and tested the changes from the 4-3-stable PR to this one - issue #2204 changes excluded for now.

**Comment by korydraughn** [comment]
What's the current status of this PR?

**Comment by JustinKyleJames** [comment]
This pull request is up-to-date.  It is a cherry-pick with necessary alterations for 4.2 - mostly just Python3 -> Python2.

Tests passed.

**Review by alanking — APPROVED** [review]
Compared with https://github.com/irods/irods_resource_plugin_s3/pull/2200 and everything seemed correct to me. # it

**Review by korydraughn — APPROVED** [review]

**Comment by JustinKyleJames** [comment]
> Compared with #2200 and everything seemed correct to me. # it

Done

**Pull request merged**